dashboard/app/models/user.rb

```diff
@@ -1646,6 +1646,39 @@ def associate_with_potential_pd_enrollments
     end
   end
 
+  # When creating an account, we want to look for any channels that got created
+  # for this user before they signed in, and if any of them are in our Applab HOC
+  # course, we will create a UserScript entry so that they get a course card
+  def generate_progress_from_storage_id(storage_id, script_name='applab-intro')
+    # applab-intro is not seeded in our minimal test env used on test/circle. We
+    # should be able to handle this gracefully
+    begin
+      script = Script.get_from_cache(script_name)
+    rescue ActiveRecord::RecordNotFound
+      nil
```

Comment:
This seems like a weird, throwaway statement.  Did you mean `script = nil`?  Do we do this for a no-op rescue - is an empty rescue block not allowed?  Can we move the return statement into the rescue block, or can script end up nil without a `ActiveRecord::RecordNotFound` being raised?

Otherwise, the early return here also feels unconventional - I'd expect all code that uses `script` to be inside this block.  I wonder if the `rescue` should be on the method definition block instead?

Reply:
I'm pretty sure I copied this pattern from somewhere else, and just assumed an empty block was not allowed for some reason. I can try removing the nil and see if things still work.

Reply:
Maybe this is more clear?

```ruby
script = begin
  Script.get_from_cache(script_name)
rescue ActiveRecord::RecordNotFound
  nil
end
```

Reply:
Alternatively it might be worth adding an optional keyword arg to `Script.get_from_cache` something like `raise_if_missing: true`, which can be supplied false just in this case

Reply:
I think I can maybe actually just do this?
```ruby
script = Script.get_from_cache(script_name) rescue nil
```

Reply:
You can, but that's generally discouraged because it will rescue anything not just a specific type of error

Reply:
TIL.  I'm cool with any of these suggestions.